File: java/344A.java
// 344 A. Magnets

import java.util.*;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        String prev = sc.next();
        int groups = 1; // first magnet always starts a group

        for (int i = 1; i < n; i++) {
            String curr = sc.next();
            if (!curr.equals(prev)) {
                groups++;
            }
            prev = curr;
        }

        System.out.println(groups);
    }
}

File: cpp/344A.cpp
// 344 A. Magnets

#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;
    string prev, curr;
    int groups = 1;

    cin >> prev;
    for (int i = 1; i < n; i++) {
        cin >> curr;
        if (curr != prev) { 
            groups++;
        }
        prev = curr; 
    }
    cout << groups;
    return 0;
}
